paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java

```diff
@@ -74,8 +74,13 @@ private TagAutoCreation(
         } else {
             Snapshot lastTag = tags.lastKey();
             this.nextSnapshot = lastTag.id() + 1;
-
-            LocalDateTime time = periodHandler.tagToTime(tags.get(lastTag));
+            LocalDateTime time;
+            String lastTagValue = tags.get(lastTag);
+            if (periodHandler instanceof TagPeriodHandler.DailyTagPeriodHandler) {
+                time = periodHandler.tagToTime(lastTagValue.split(" ")[0]);
+            } else {
+                time = periodHandler.tagToTime(lastTagValue);
+            }
```

Comment:
You can just place the `split` logic into `DailyTagPeriodHandler#tagToTime`.

Reply:
done.